### ISSUE TITLE
warn  - `enableUndici` option is unnecessary in Node.js v18.0.0 or greater. 

### ISSUE THREAD

**Issue by BWBama85** [vercel/next.js#42441]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
    Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 18.12.0
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.0.2-canary.2
      eslint-config-next: 13.0.1
      react: 18.2.0
      react-dom: 18.2.0
```

### Describe the Bug

When running npm run dev I get:

warn  - `enableUndici` option is unnecessary in Node.js v18.0.0 or greater. 

My next.config.js:

```
/** @type {import('next').NextConfig} */
const nextConfig = {
  experimental: {
    appDir: true,
  },
};

module.exports = nextConfig;
```

### Expected Behavior

Not to see this error message. I do not have this option in my config. 

### Link to reproduction

n/a

### To Reproduce

![next](https://user-images.githubusercontent.com/41128321/199828947-0c3cadf7-df8b-4835-b320-ad7c7f21c4bb.png)

**Comment by ijjk** [comment]
Hi this should be updated in `v13.0.2` please update and give it a try!

**Comment by BWBama85** [comment]
Still seeing it. 

```
    Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 18.12.0
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.0.2
      eslint-config-next: 13.0.2
      react: 18.2.0
      react-dom: 18.2.0
```

This time I do not get the warn immediately after starting, but while loading pages. 

Initial start:
```
PS C:\Users\bamas\OneDrive\Documents\My Backups\Development\nodejs\thewilsonnet.com> npm run dev

> thewilsonnet.com@0.1.0 dev
> next dev

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
warn  - You have enabled experimental feature (appDir) in next.config.js.
warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.
info  - Thank you for testing `appDir` please leave your feedback at https://nextjs.link/app-feedback

event - compiled client and server successfully in 665 ms (200 modules)
```
After loading a page:
```
PS C:\Users\bamas\OneDrive\Documents\My Backups\Development\nodejs\thewilsonnet.com> npm run dev

> thewilsonnet.com@0.1.0 dev
> next dev

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
warn  - You have enabled experimental feature (appDir) in next.config.js.
warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.
info  - Thank you for testing `appDir` please leave your feedback at https://nextjs.link/app-feedback

event - compiled client and server successfully in 665 ms (200 modules)
wait  - compiling /about/resume/page (client and server)...
event - compiled client and server successfully in 904 ms (395 modules)
warn  - `enableUndici` option is unnecessary in Node.js v18.0.0 or greater.
wait  - compiling /page (client and server)...
warn  - `enableUndici` option is unnecessary in Node.js v18.0.0 or greater.
event - compiled client and server successfully in 388 ms (399 modules)
warn  - `enableUndici` option is unnecessary in Node.js v18.0.0 or greater.
warn  - `enableUndici` option is unnecessary in Node.js v18.0.0 or greater.
```

**Comment by zenflow** [comment]
I just upgraded to 13.0.2 (I had to do this because I just upgraded node from 16.x to 18.12.0 and needed the fix from #42382) 

I am seeing the warnings during `next build` (it's mangled with the animated console output):

```
warn  - Linting is disabled
warn  - You have enabled experimental feature (appDir) in next.config.js.
warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.
info  - Thank you for testing `appDir` please leave your feedback at https://nextjs.link/app-feedback

info  - Creating an optimized production build
info  - Compiled successfully
info  - Checking validity of types .info  - VS Code settings.json has been updated for Next.js' automatic app types, this file can be added to .gitignore if desired
info  - Checking validity of types
info  - Collecting page data .warn  - `enableUndici` option is unnecessary in Node.js v18.0.0 or greater.
info  - Collecting page data
[=   ] info  - Generating static pages (0/3)warn  - `enableUndici` option is unnecessary in Node.js v18.0.0 or greater.
warn  - `enableUndici` option is unnecessary in Node.js v18.0.0 or greater.
[==  ] info  - Generating static pages (0/3)warn  - `enableUndici` option is unnecessary in Node.js v18.0.0 or greater.
info  - Generating static pages (3/3)
info  - Finalizing page optimization
```

**Comment by zenflow** [comment]
@ooyay They fixed it in v13.0.3-canary.0

See https://github.com/vercel/next.js/releases/tag/v13.0.3-canary.0

You can use `13.0.3-canary.0` in your package.json or wait for the next official release (13.0.3)
